src/util/tracking.ts

```diff
@@ -128,37 +131,54 @@ export function trackError(
 /**
  * Send a raw event to all backends.
  */
-export function logEvent(event: TrackingEventName, values: TrackingValues = {}) {
-  const { accountDate, currencyCode, dollarValue, installerId, pluginId, error } = values
-  getExperimentConfig()
-    .then(async (experimentConfig: ExperimentConfig) => {
-      // Persistent & Unchanged params:
-      const { isFirstOpen, deviceId, firstOpenEpoch } = await getFirstOpenInfo()
-      const params: any = { edgeVersion: getVersion(), isFirstOpen, deviceId, firstOpenEpoch, ...values }
-
-      // Adjust params:
-      if (accountDate != null) params.adate = accountDate
-      if (currencyCode != null) params.currency = currencyCode
-      if (dollarValue != null) {
-        params.currency = 'USD'
-        params.value = Number(dollarValue.toFixed(2))
-        params.items = [String(event)]
-      }
-      if (installerId != null) params.aid = installerId
-      if (pluginId != null) params.plugin = pluginId
-      if (error != null) params.error = makeErrorLog(error)
-
-      // Add all 'sticky' remote config variant values:
-      for (const key of Object.keys(experimentConfig)) params[`svar_${key}`] = experimentConfig[key as keyof ExperimentConfig]
-
-      // TEMP HACK: Add renamed var for legacyLanding
-      params.svar_newLegacyLanding = experimentConfig.legacyLanding
-
-      consify({ logEvent: { event, params } })
-
-      Promise.all([logToPosthog(event, params), logToFirebase(event, params), logToUtilServer(event, params)]).catch(error => console.warn(error))
-    })
-    .catch(console.error)
+export function logEvent(event: TrackingEventName, values: TrackingValues = {}): ThunkAction<void> {
+  return async (dispatch, getState) => {
+    const { currencyCode, dollarValue, pluginId, error, exchangeAmount } = values
+    getExperimentConfig()
+      .then(async (experimentConfig: ExperimentConfig) => {
+        // Persistent & Unchanged params:
+        const { isFirstOpen, deviceId, firstOpenEpoch } = await getFirstOpenInfo()
+
+        const params: any = { edgeVersion: getVersion(), buildNumber: getBuildNumber(), isFirstOpen, deviceId, firstOpenEpoch, ...values }
+
+        // Populate referral params:
+        const state = getState()
+        const { deviceReferral, account } = state
+        const { accountReferral } = account
+        params.refDeviceId = deviceReferral.installerId
+        params.refDeviceCcs = deviceReferral.currencyCodes
+
+        const { creationDate, installerId } = accountReferral
+        params.refAccountDate = installerId == null || creationDate == null ? undefined : creationDate.toISOString().replace(/-\d\dT.*/, '')
```

Comment:
We shouldn't be reporting the device installerId. It's only relevant if we don't have an account installerId. So really we should only have

```
params.installerId = accountReferral.installerId ?? deviceReferral.installerId
params.refCurrencyCodes = accountReferral.currencyCodes ?? deviceReferral. currencyCodes
params.creationDate = accountReferral.creationDate
```

Reply:
We need this to report ID's prior to login

Reply:
After a lot of back & forth, we did agree to track both values separately (Jon, William, and Peter). I agree with Paul that it seems unnecessary at first glance, but after thinking about it more deeply, separate tracking is in fact the correct solution.

Reply:
ok. That's fair.  Can I just request a simple rename as I don't think the current codes are properly represented, especially the term `deviceId` since it is definitely NOT a deviceId, and we can't change them later without breaking the reporting tools. 

I'd ask for

```
refDeviceInstallerId
refDeviceCurrencyCodes
refAccountInstallerId
refAccountCurrencyCodes
refAccountDate
```